### PR TITLE
Guard against `InaccessibleObjectException`s

### PR DESCRIPTION
**Pull request by vemv**
Fixes https://github.com/nrepl/nrepl/issues/263

### CODE REVIEW

**Comment by vemv** [comment]
mmm InaccessibleObjectExceptions doesn't exist under JDK8, will have to tweak.

**Comment by vemv** [comment]
PR ready. Unfortunately I never got to reproduce the issue in question.

It's weird: our CI has JDK17 and I know from various sources that nrepl's reflective code throws a `InaccessibleObjectException`.

Maybe it depends on how the JDK itself was built, idk.

**Comment by bbatsov** [comment]
I'll have to think more about this, as I'm quite puzzled by it. We're guarding towards something that doesn't happen in our own CI? I also don't quite get what the fallback achieves other than catching the exception. I mean you still the need the reflection to achieve what the code needs to do.

File: src/clojure/nrepl/middleware/interruptible_eval.clj [hunk 1]
```diff
@@ -39,11 +39,15 @@
         (doto (.setAccessible true))
         (.set reader column))))
 
-(defn- source-logging-pushback-reader
+(defn source-logging-pushback-reader
```
Comment (bbatsov) [inline src/clojure/nrepl/middleware/interruptible_eval.clj:42]: My make this public? It is just to test it?

File: src/clojure/nrepl/middleware/interruptible_eval.clj [hunk 2]
```diff
@@ -39,11 +39,15 @@
...
   [code line column]
   (let [reader (LineNumberingPushbackReader. (StringReader. code))]
-    (when line (set-line! reader (int line)))
-    (when column (set-column! reader (int column)))
+    (try
```
Comment (bbatsov) [inline src/clojure/nrepl/middleware/interruptible_eval.clj:45]: This and the other similar instance should have some comments explaining the need for the try/catch, as it's definitely not obvious.

**Comment by alexander-yakushev** [comment]
This can be shelved until we have to access something from `java.base` or any other module through reflection. #263 is already closed as Fixed.

**Pull request closed without merging**